src/sql/azdata.proposed.d.ts

```diff
@@ -136,6 +136,13 @@ declare module 'azdata' {
 		loadingCompletedText?: string;
 	}
 
+	export interface Component extends ComponentProperties {
+		/**
+		 * Dispose the component and clean up its resources like listeners and do
+		 */
+		destroy(): void;
```

Comment:
Why not just make these components disposable? That's what @cheenamalhotra is doing and fits better into the existing functionality/expectations.